### PR TITLE
Passphrase flow checks device for empty and not for visible accounts

### LINKED ISSUE #14208: Adding an empty passphrase don't prompt me to check my passphrase
**Describe the bug**
When I add a new/empty passphrase, I should be prompted/shown that the passphrase is empty and if I want to continue and I should re-enter it but I am not on latest develop

**Info:**

-   Mobile Suite Version: 24.9.1 b415aa2
-   OS: Android

**How to reproduce**

1. Connect a device
2. Go to passphrase
3. Enter an empty passphrase
4. It goes straight to discovery (without confirming it is empty)

**Expected behavior**
I should be prompted that the passphrase is empty and I should re-enter it.

### PR DESCRIPTION
**Pull request by vytick**
- Passphrase flow previously checked accounts visibility, now it checks if account is empty instead.
- I also renamed `selectIsEmptyDevice` to `selectIsDiscoveredAccountlessDevice` because that is what it actually returns.

Resolve #14208

## Screenshots:
https://github.com/user-attachments/assets/300c2a20-f02b-4d33-b966-a8c13cbfa900

### CODE REVIEW

**Review by yanascz — APPROVED** [review]
Seems to work fine!

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 1]
```diff
@@ -397,9 +401,23 @@ export const selectIsEmptyDevice = (
...
         const hasOnlyPortfolioDevice = selectHasOnlyPortfolioDevice(state);
 
-        return isEmptyDevice && hasOnlyPortfolioDevice;
+        return isDiscoveredAccountlessDevice && hasOnlyPortfolioDevice;
+    },
+);
+
+// If we know that device is not empty, we can return true right away.
+// Otherwise we return null and wait until discovery finishes
+export const selectDeviceIsNotEmpty = memoize(
```
Comment (matejkriz) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts]: It returns `boolean` or `null`, right? I see no point of memoizing this. And I would change the name:
```suggestion
export const selectIsDeviceNotEmpty = (
```

nitpick: I find `if` conditions without brackets unnecessary harder to read, especially if the condition is longer and it's not just `if(isLoading) return null;` or something like that... But feel free to disagree :-)

File: suite-common/wallet-core/src/accounts/accountsReducer.ts [hunk 2]
```diff
@@ -386,7 +390,7 @@ export const selectIsDeviceAccountless = (state: AccountsRootState & DeviceRootS
     pipe(selectVisibleDeviceAccounts(state), A.isEmpty);
 
 // Selected device has no accounts and no active discovery. It can be empty portfolio device.
-export const selectIsEmptyDevice = (
+export const selectIsDiscoveredAccountlessDevice = (
```
Comment (matejkriz) [inline suite-common/wallet-core/src/accounts/accountsReducer.ts]: nitpick, feel free to ignore
```suggestion
export const selectIsDiscoveredDeviceAccountless = (
```

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 1 above)
Comment (vytick) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts]: I actually do feel the same and dont know, why I havent used those. I'll update and remove memoization

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 2 above)
Comment (vytick) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts]: e6791b5ea8

File: suite-common/wallet-core/src/accounts/accountsReducer.ts (hunk 1 above)
Comment (vytick) [reply suite-common/wallet-core/src/accounts/accountsReducer.ts]: e6791b5ea8

**Review by matejkriz — APPROVED** [review]

**Pull request merged**